runtime/lua/vim/lsp/document_color.lua

```diff
@@ -0,0 +1,84 @@
+local document_color_ns = vim.api.nvim_create_namespace('nvim.lsp.document_color')
+
+local M = {}
+
+--- @alias vim.lsp.document_color.HighlightInfo { hl_group: string, range: Range4 }
+
+--- @type table<string, true>
+local color_cache = {}
+
+--- @param color lsp.Color
+--- @return string
+local function get_hl_group(color)
```

Comment:
does this function can be public 🤔 then  https://github.com/neovim/neovim/pull/32138 can reuse it in generate_kind..

Reply:
Sure! Let me finish the implementation first and then we can refactor it to make some of these utilities public :)

Reply:
thank you